### PR TITLE
Added support for Entity Framework 6+ DbContext driver

### PR DESCRIPTION
**Pull request by SteveRuble**
This closes issue #4 by adding support for query plan dumping when using an Entity Framework DbContext rather than LINQ to SQL.

### CODE REVIEW

**Comment by Giorgi** [comment]
@SteveRuble Thanks for the PR. LINQPad supports typed data context for both LINQ to SQL and EF. Does this solve the issue with both cases?

File: src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                Connection.Open();
+
+                try
+                {
+                    using (var command = Connection.CreateCommand())
+                    {
+                        command.CommandText = "SET STATISTICS XML ON";
+                        command.ExecuteNonQuery();
+                    }
+
```
Comment (Giorgi) [inline src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs]: I don't like that this command exists in two different classes. I think you can move the common code to base classes and only override the part which needs to get the actual command.

File: src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                        foreach (DbParameter parameter in interceptor.Command.Parameters)
+                        {
+                            var parameterCopy = command.CreateParameter();
+                            parameterCopy.ParameterName = parameter.ParameterName;
+                            parameterCopy.DbType = parameter.DbType;
+                            parameterCopy.Value = parameter.Value;
+
+                            command.Parameters.Add(parameterCopy);
+                        }
+
```
Comment (Giorgi) [inline src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs]: I think it's better to use AddRange and get rid of the foreach loop.

File: src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            public void ScalarExecuted(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
+            {
+            }
+        }
+
+        public Task CreateIndexAsync(DbConnection connection, string script)
+        {
+
+            return null;
+        }
```
Comment (Giorgi) [inline src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs]: Is this needed at all?

Comment (SteveRuble) [inline src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs]: Nope. I'll delete it.

File: src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs (hunk 1 above)
Comment (SteveRuble) [inline src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs]: Yep, you're right. All the derived class really needs to do is provide the DbCommand for the base class to run. It's a lot cleaner now.

**Comment by Giorgi** [comment]
@SteveRuble Does this support typed context for LINQ to SQL too?

**Comment by SteveRuble** [comment]
Yes. The typed context sets Util.CurrentDataContext the same way the dynamic context does, so it works the same way with the DatabaseHelper.

(Sorry, I thought I answered this the first time you asked.)

**Comment by Giorgi** [comment]
@SteveRuble Merged. Tested and works with EF. Will it work with typed Linq to sql context ?  I don't think the DbInterception will be able to capture LINQ to SQL queries.

**Comment by SteveRuble** [comment]
The typed Linq to SQL context sets Util.CurrentDataContext the same way the dynamic Linq to SQL context does, so it is handled by the LinqToSqlDatabaseHelper.

**Comment by SteveRuble** [comment]
When I tested a compiled LINQ to SQL query it set Util.CurrentDataContext to the compiled DataContext, but I see you've checked in code targeting the compiled LINQ to SQL so maybe I was doing something non-standard.

**Pull request merged**